src/Widgets/calendar/CalendarView.vala

```diff
@@ -79,6 +79,31 @@ public class DateTime.Widgets.CalendarView : Gtk.Grid {
         right_grid.update_weeks (calmodel.data_range.first_dt, calmodel.num_weeks);
         calmodel.change_month (-1);
 
+        CalendarModel.get_default ().locale_settings.changed.connect (() => {
+            // If First Day setting changes, reset every grid.
+            carousel.no_show_all = true;
```

Comment:
Better to construct `carousel` before using it :stuck_out_tongue:

Reply:
There are more problems with this PR, but I think those are out of scope and could be handled in a newer PR.

But thanks for finding this.